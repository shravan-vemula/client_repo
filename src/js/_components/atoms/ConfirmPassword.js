import React,{useState} from 'react'
import TextField from '@material-ui/core/TextField';
import ClearOutlinedIcon from '@material-ui/icons/ClearOutlined';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';

function ConfirmPassword(props) {
    var [colorValue, handleValidity] = useState("secondary")
    var [passwordMatchInValidity, handlePasswordInvalidity] = useState("validity-text valid")
    var [passwordMatchValidity, handlePasswordValidity] = useState("validity-text valid")
    

    

    function ConfirmPasswordEvent(event){

        const password = event.target.value;
        console.log(password)

        if(document.getElementById("primary-password-filled-secondary").value == password){
            handlePasswordValidity("validity-text valid confirmed")
            handlePasswordInvalidity("validity-text valid")
            handleValidity("primary")
            props.state.confirmPassword = true
            console.log("state set to "+props.state.error)
        }
        else{
            handlePasswordValidity("validity-text valid ")
            handlePasswordInvalidity("validity-text ")
            props.state.confirmPassword = false
        }


    }
    return (
        <React.Fragment>
        <div className="password-input-field"style={{marginTop: "20px"}}>
        <TextField required 
        id="standard-password-input" 
        label="Confirm Password"
        type="password"
        variant="outlined"
        size="small"
        inputProps={{ "data-testid": "confirmPassword" }}
        color={colorValue}
        fullWidth
        onBlur={(event)=>{ConfirmPasswordEvent(event)}}
        />
        </div>
        <p className={passwordMatchInValidity}><span id="cross" className="validity-text cross"><ClearOutlinedIcon /></span>
            Does not match with given password
        </p>
        <p className={passwordMatchValidity}><span id="done" className="validity-text accepted success"><CheckCircleIcon /></span>
            Matches with the given password
        </p>
        </React.Fragment>
    )
}

export default ConfirmPassword
