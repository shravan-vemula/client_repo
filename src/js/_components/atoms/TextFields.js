import React, { useState } from 'react'
import TextField from '@material-ui/core/TextField';




function TextFields(props) {


    console.log("props state text field " + props.state)

    var [className,setClassName] = useState("CheckValidity");

    function validateEmailId(email){
        const emailPattern = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
        if(email.length == 0){

            props.state.errorEmail = false
            return
        }
        if (emailPattern.test(email))
        {
            setClassName("CheckValidity");
            props.state.errorEmail = true
            return;
        }
        console.log("false");
        setClassName("displayBlock");
        props.state.errorEmail = false
    }

    function validateEntry(event){
        if(props.useCase == "EmailId"){
            
            
            validateEmailId(event.target.value);
        }
    }

    console.log(props.useCase)
    return (
    <React.Fragment>
        <div className="form-input-field" style={{marginTop: "20px"}}>
        <TextField required 
        id="standard-required-emailId" 
        label= {props.value}
        inputProps={{ "data-testid": "textField" }}
        variant="outlined"
        size="small"
        fullWidth
        onBlur= {validateEntry}
       />
        </div>
        <div id="check-validity" className={className}>
        <p className="validity-text length">Enter a Valid {props.useCase}</p>
            </div>

    </React.Fragment>
    )
}

export default TextFields;