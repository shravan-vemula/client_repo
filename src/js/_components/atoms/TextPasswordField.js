import React,{useState} from 'react'
import TextField from '@material-ui/core/TextField';
import ClearOutlinedIcon from '@material-ui/icons/ClearOutlined';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';




function TextPasswordField(props) {




    
    var [classValue, handleDisplayCases] = useState("passwordValidity")
    var [colorValue, handleValidity] = useState("secondary")
    var [lenghtValidity, handleLength] = useState("validity-text valid")
    var [caseValidity, handleCase] = useState("validity-text valid")
    var [numberValidity, handleNumber] = useState("validity-text valid")
    var [symbolValidity, handleSymbol] = useState("validity-text valid")
    var [numberInValidity,handleNumberInvalidity] = useState("validity-text")
    var [SymbolInValidity,handleSymbolInvalidity] = useState("validity-text")
    var [caseInValidity,handleCaseInvalidity] = useState("validity-text")
    var [lengthInValidity,handleLengthInvalidity] = useState("validity-text")
    var [strongValidity,handleStrongValidity] = useState("strongEntry valid") 

    // function handleChange(){
    //     console.log(document.getElementsByClassName("passwordValidity").passwordValidityId.style.display);
    //     document.getElementsByClassName("passwordValidity").passwordValidityId.style.display+="none !important";
    // }
    


    console.log("props state " + props.state)

    function handleChange(event){
        let count = 0;
        handleDisplayCases("passwordValidity displayBlock");
        let validity = false
        let password = event.target.value
        if(password.length >= 8){
            handleLength("validity-text valid length")
            handleLengthInvalidity("validity-text valid")
            console.log("length checked")
            props.state.errorPassword = true
            validity = true
            count++;
        }
        else
        {
            console.log("length not checked")
            handleLength("validity-text valid")
            handleLengthInvalidity("validity-text")
            props.state.errorPassword = false
            validity = false
            count--;
        }

        if(password.match(/[a-z]/g) && password.match(/[A-Z]/g)){
            console.log("Cases checked")
            handleCase("validity-text valid length")
            handleCaseInvalidity("validity-text valid")
            props.state.errorPassword = true
            validity = true
            count++;
        }
        else
        {
            console.log("Cases not checked")
            handleCase("validity-text valid")
            handleCaseInvalidity("validity-text")
            props.state.errorPassword = false
            validity = false
            count--;
        }


        if(password.match(/[0-9]/g)){
            console.log("number checked")
            handleNumber("validity-text valid length")
            handleNumberInvalidity("validity-text valid")
            props.state.errorPassword = true
            validity = true
            count++;
        }
        else
        {
            console.log("number not checked")
            handleNumber("validity-text valid")
            handleNumberInvalidity("validity-text")
            props.state.errorPassword = false
            validity = false
            count--;
        }


        if(password.match(/[^a-zA-Z\d]/g)){
            handleSymbol("validity-text valid length")
            handleSymbolInvalidity("validity-text valid")
            props.state.errorPassword = true
            validity = true
            count++;

        }
        else
        {
            handleSymbol("validity-text valid")
            handleSymbolInvalidity("validity-text")
            props.state.errorPassword = false
            validity = false
            count--;
        }

        console.log(count)
        if(validity && count ==4){
            handleDisplayCases("passwordValidity")
            handleValidity("primary")
            handleStrongValidity("strongEntry success validity-text")
        }
        else{
            props.state.errorPassword = false
            handleDisplayCases("validity-text")
            handleStrongValidity("strongEntry valid")
        }

    }


    

    
    return (
        <React.Fragment>
        <div className="password-input-field"style={{marginTop: "20px"}}>
        <TextField required 
        id="primary-password-filled-secondary" 
        label={props.value}
        color={colorValue}
        type="password"
        variant="outlined"
        inputProps={{ "data-testid": "passwordField" }}
        size="small"
        fullWidth
        onChange= {(event)=>{handleChange(event)}}
        />

        <div className={classValue} id="passwordValidityId">
                        <p className={lengthInValidity}><span id="cross" className="validity-text cross"><ClearOutlinedIcon /></span>
                        Use 8 or more characters
                        </p>
                        <p className={lenghtValidity}><span id="done" className="validity-text accepted success"><CheckCircleIcon /></span>
                        Use 8 or more characters
                        </p>
                        <p className={caseInValidity}><span id="cross" className="validity-text cross"><ClearOutlinedIcon /></span>
                            Use upper and lower case letters (e.g. Aa) </p>
                            <p className={caseValidity}><span id="done" className="validity-text accepted success"><CheckCircleIcon /></span>
                            Use upper and lower case letters (e.g. Aa) </p>
                        <p className={numberInValidity}>
                        <span id="cross" className="validity-text cross"><ClearOutlinedIcon /></span>
                        Use a number (e.g. 1234)</p>
                        <p className={numberValidity}>
                        <span id="done" className="validity-text accepted success"><CheckCircleIcon /></span>
                        Use a number (e.g. 1234)</p>
                        <p className={SymbolInValidity}>
                        <span id="cross" className="validity-text cross"><ClearOutlinedIcon /></span>
                        Use a symbol (e.g. !@#$) </p> 
                        <p className={symbolValidity}>
                        <span id="done" className="validity-text accepted success"><CheckCircleIcon /></span>
                        Use a symbol (e.g. !@#$) </p> 
                    </div>

        
            <div className={strongValidity} id="strong-password">
                <p className="strong validity-text"> Password is strong </p>
            </div>
      
      
        </div>
    </React.Fragment>
    )
    
}


export default TextPasswordField
