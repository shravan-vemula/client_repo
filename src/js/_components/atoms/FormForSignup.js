import React, { Component } from "react";
import TextFields from "./TextFields";
import TextPasswordField from "./TextPasswordField";
import { Button } from "@material-ui/core";
import ConfirmPassword from "./ConfirmPassword";
import LockIcon from "@material-ui/icons/Lock";
import { Link } from "react-router-dom";

export class FormForSignup extends Component {
  constructor(props) {
    super(props);

    this.state = {
      errorEmail: false,
      errorPassword: false,
      confirmPassword: false,
    };
  }

  handleSubmit(event) {
    if (
      !(
        this.state.errorEmail &&
        this.state.errorPassword &&
        this.state.confirmPassword
      )
    ) {
      console.log(
        this.state.errorEmail +
          "       " +
          this.state.errorPassword +
          "        " +
          this.state.confirmPassword
      );
      alert("enter valid details");
      event.preventDefault();
      return;
    } else {
      const email = document.getElementById("standard-required-emailId").value;
      console.log("email " + email);
    }
  }

  render() {
    return (
      <React.Fragment>
        <div id="FormForSignup" className="signup-form">
          <form
            data-testId="formSubmit"
            onSubmit={(event) => {
              this.handleSubmit(event);
            }}
          >
            <TextFields
              value="Email Address"
              useCase="EmailId"
              state={this.state}
            />
            <TextPasswordField
              value="Password"
              state={this.state}
            ></TextPasswordField>
            <ConfirmPassword state={this.state} />
            {/* <TextPasswordField value="confirm password" /> */}
            <div className="submitButton">
              <Link to={"/SelectBank/BankTiles"}>
                {" "}
                <Button
                  onClick={(e) => this.handleSubmit(e)}
                  data-testId="submitButton"
                  variant="contained"
                  color="primary"
                >
                  <span id="lockIcon">
                    <LockIcon />
                  </span>{" "}
                  Create account
                </Button>
              </Link>
            </div>
          </form>
        </div>
      </React.Fragment>
    );
  }
}

export default FormForSignup;
