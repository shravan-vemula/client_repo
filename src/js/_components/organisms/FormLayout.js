import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import HeaderForForm from '../atoms/HeaderForForm';
import FormForSignup from '../atoms/FormForSignup';
import AlreadySignedIn from '../atoms/AlreadySignedIn';
import 'js/_components/form.css';
import 'js/_components/sign-up.css';
import logo from "images/Btracker.png";


function FormLayout() {
  return (
    <React.Fragment>
       <AlreadySignedIn />
    <div className="form-layout" style={{fontFamily:"Inter,sans-serif"}}>
      <CssBaseline />
     
 
      <div id="budget-tracker-heading" align="center" className="budgetTrackerHeading" style={{fontFamily:"Inter,sans-serif"}}>
        <img src={logo} alt="Logo" height="100%" width="80%" ></img>
        </div>

      <Container maxWidth="sm" className="form-div">

        <Typography component="div" style={{ height: 'auto',marginBottom: "10%"}} style={{fontFamily:"Inter,sans-serif"}}>
        <HeaderForForm heading='Create an Account' />
        <FormForSignup />
        </Typography>
      </Container>

    </div>
    </React.Fragment>
  );
}

export default FormLayout;