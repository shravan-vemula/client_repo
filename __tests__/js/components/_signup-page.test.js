import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import App from 'src/App.js';
import FormLayout from 'js/_components/organisms/FormLayout';
import {BrowserRouter as Router,Switch,Route} from 'react-router-dom';
import FormForSignup from 'js/_components/atoms/FormForSignup';
import TextFields from 'js/_components/atoms/TextFields';
import TextPasswordField from 'js/_components/atoms/TextPasswordField';
import ConfirmPassword from 'js/_components/atoms/ConfirmPassword';
import AnimantionModalSuccess from 'js/_components/atoms/AnimationModalSuccess';

afterEach(cleanup)

it('renders without crashing',() => {
    const div = document.createElement('div');
    ReactDOM.render(<App />,div);
})

describe('FormLayout', () =>{
    test('renders form layout', () =>{
        render(<Router><FormLayout /></Router>);
        expect(screen.getByText('Create an Account')).toBe;
        expect(screen.getByText('Already have an account?')).toBe;
        expect(screen.getByRole('img')).toBe;
    })
})

describe('FormLayout', () =>{

    it("renders confirm Password field correctly", () => {
        const wrapper = shallow(
            <FormLayout /> 
        );
        expect(wrapper).toMatchSnapshot();
      });

    it("Form renders correctly", () => {
        const wrapper = shallow(
            <FormForSignup />
        );
        expect(wrapper).toMatchSnapshot();
      });

      it("renders Email text box correctly", () => {
        const wrapper = shallow(
            <TextFields value='Email Address' useCase="EmailId" />
        );
        expect(wrapper).toMatchSnapshot();
      });

      it("renders Password field correctly", () => {
        const wrapper = shallow(
            <TextPasswordField value="Password" /> 
        );
        expect(wrapper).toMatchSnapshot();
      });


      it("renders confirm Password field correctly", () => {
        const wrapper = shallow(
            <ConfirmPassword /> 
        );
        expect(wrapper).toMatchSnapshot();
      });
    

    test('renders form layout', () =>{
        render(<Router><FormForSignup /></Router>);
        expect(screen.getByText('Email Address')).toBe;
        expect(screen.getByText('Password')).toBe;
        expect(screen.getByText('Confirm Password')).toBe;
        expect(screen.getByRole('button')).toBe;
    })

    


    it("should have textfield and validations are implemented",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("textField")).toBe;
        fireEvent.blur(getByTestId("textField"), {
          target: { value: "@gmail.com" }
        });
        expect(screen.getByText('Enter a Valid EmailId')).toBe;
    });

    it("should have textfield and validations are implemented",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("textField")).toBe;
        fireEvent.blur(getByTestId("textField"), {
          target: { value: "kausik@gmail.com" }
        });
    });

    it("should have textfield and validations are implemented",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("textField")).toBe;
        fireEvent.blur(getByTestId("textField"), {
          target: { value: " " }
        });
        expect(screen.getByText('Enter a Valid EmailId')).toBe;
    });

    it("should have textfield and validations are implemented",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("textField")).toBe;
        fireEvent.blur(getByTestId("textField"), {
          target: { value: "" }
        });
    });

    it("password valid",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("passwordField")).toBe;
        fireEvent.blur(getByTestId("passwordField"), {
            target: { value: "Theracer2!" }
        });
        screen.debug();
        expect(screen.getByText('Password is strong')).toBe;
    });

    it("should have confirm password",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("confirmPassword")).toBe;
        fireEvent.blur(getByTestId("confirmPassword"), {
            target: { value: "com" }
        });
        expect(screen.getByText('Does not match with given password')).toBe;
    });

    it("password strength validation",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("passwordField")).toBe;
        fireEvent.change(getByTestId("passwordField"), {
            target: { value: "com" }
        });
        
    });


    it("password and confirm password Match",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("passwordField")).toBe;
        fireEvent.change(getByTestId("passwordField"), {
            target: { value: "Theracer2!" }
        });
        expect(getByTestId("confirmPassword")).toBe;
        fireEvent.change(getByTestId("confirmPassword"), {
            target: { value: "Theracer2!" }
        });
        expect(screen.getByText('Password is strong')).toBe;
        expect(screen.getByText('Matches with the given password')).toBe;
    });

    it("password and confirm password Match",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("passwordField")).toBe;
        fireEvent.blur(getByTestId("passwordField"), {
            target: { value: "Theracer2!" }
        });
        expect(getByTestId("confirmPassword")).toBe;
        fireEvent.blur(getByTestId("confirmPassword"), {
            target: { value: "Theracer2!" }
        });
        expect(screen.getByText('Password is strong')).toBe;
        expect(screen.getByText('Does not match with given password')).toBe;
    });

    it("Form Submit",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("textField")).toBe;
        fireEvent.blur(getByTestId("textField"), {
          target: { value: "kausik@gmail.com" }
        });
        
        expect(getByTestId("passwordField")).toBe;
        fireEvent.blur(getByTestId("passwordField"), {
            target: { value: "Theracer2!" }
        });
        expect(getByTestId("confirmPassword")).toBe;
        fireEvent.blur(getByTestId("confirmPassword"), {
            target: { value: "Theracer2!" }
        });


        expect(getByTestId("formSubmit")).toBe;
        fireEvent.submit(getByTestId("formSubmit"))

        expect(getByTestId("submitButton")).toBe;
        fireEvent.click(getByTestId("submitButton"))




    });

    it("Form Submit error",() =>{
        const { getByTestId } =  render(<Router><FormForSignup /></Router>);
        expect(getByTestId("textField")).toBe;
        fireEvent.blur(getByTestId("textField"), {
          target: { value: "@gmail.com" }
        });
        expect(getByTestId("passwordField")).toBe;
        fireEvent.change(getByTestId("passwordField"), {
            target: { value: "Theracer2!" }
        });
        expect(getByTestId("confirmPassword")).toBe;
        fireEvent.change(getByTestId("confirmPassword"), {
            target: { value: "Theracer2!" }
        });

        expect(getByTestId("formSubmit")).toBe;
        fireEvent.submit(getByTestId("formSubmit"))

        expect(getByTestId("submitButton")).toBe;
        fireEvent.click(getByTestId("submitButton"))

    });


    

})


