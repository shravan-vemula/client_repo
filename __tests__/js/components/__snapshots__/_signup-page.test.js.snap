// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLayout Form renders correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders Email text box correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders Password field correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders confirm Password field correctly 1`] = `ShallowWrapper {}`;

exports[`FormLayout renders confirm Password field correctly 2`] = `ShallowWrapper {}`;
